### ISSUE TITLE
Terraform state commands not working for modules with Azure backend

### ISSUE THREAD

**Issue by phatcher** [hashicorp/terraform#27510]
### Terraform Version

```
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/azuread v1.0.0
+ provider registry.terraform.io/hashicorp/azurerm v2.41.0  
```

### Expected Behavior

I have a terraform configuration using an Azure backend which works fine.

However, if I try to use `terraform state` commands against anything other than exact addresses, the terraform cannot find the information.

### Actual Behavior

If I issue a `terraform state list` I see the following

```
module.key_vault["Dev"].azurerm_key_vault.kv
module.key_vault["DevInt"].azurerm_key_vault.kv     
module.key_vault["DevTest"].azurerm_key_vault.kv    
module.key_vault["Integration"].azurerm_key_vault.kv
module.key_vault["Test"].azurerm_key_vault.kv
```

If I then issue `terraform state list module.key_vault` I get

```
Error: Unknown module

The current state contains no module at module.key_vault. If you've just added
this module to the configuration, you must run "terraform apply" first to
create the module's entry in the state.

```

The [state command](https://www.terraform.io/docs/commands/state/list.htmll) says this should be possible unless I'm mis-reading it somehow?

Similar things happen if I want to issue a `state mv` or `state rm`

**Comment by mildwonkey** [comment]
Hi @phatcher, thanks for reporting this !
I can see from the output that you have modules created with for_each. You need to include the specific instance of the module for state commands, so for e.g. to list the resources in the `Dev` module, you'd run `terraform state list module.key_vault["Dev"]` 
Can you let me know if this works for you?

**Comment by phatcher** [comment]
Yes, that works, but only lists the resource immediately below, the full state list is the following...

```
module.key_vault["Dev"].azurerm_key_vault.kv
module.key_vault["DevInt"].azurerm_key_vault.kv
module.key_vault["DevTest"].azurerm_key_vault.kv
module.key_vault["Integration"].azurerm_key_vault.kv
module.key_vault["Test"].azurerm_key_vault.kv
module.key_vault["Dev"].module.admins_access["xxxx"].azurerm_key_vault_access_policy.this
module.key_vault["DevInt"].module.admins_access["xxx"].azurerm_key_vault_access_policy.this
module.key_vault["DevTest"].module.admins_access["xxx"].azurerm_key_vault_access_policy.this
module.key_vault["Integration"].module.admins_access["xxx"].azurerm_key_vault_access_policy.this
module.key_vault["Test"].module.admins_access["xxx"].azurerm_key_vault_access_policy.this
```

and ```terraform state list module.key_vault["Dev"]``` only produces 

```
module.key_vault["Dev"].azurerm_key_vault.kv
```

From the documentation it says...

```
Example: Filtering by Resource
This example will only list resources for the given name:

$ terraform state list aws_instance.bar
aws_instance.bar[0]
aws_instance.bar[1]
```

So I would expect for_each and nested resources to work the same way.

I'm having to move a lot of resources around and the ability to do ```state mv module.key_vault.admins_access  module.key_vault.admins``` or ```state rm module,key_vault``` would be very helpful.  So if this isn't something the language currently supports, please take this as a feature request :smile:

At the moment I'm having to do a lot of copy/paste into an editor to get all the commands written and it makes it more difficult to refactor where things live in terraform state files, i.e. I can't afford to tidy up some of my modules since the state mvs would be so tedious/error-prone.

I could live with ```terraform state mv module.key_vault["Dev"].module.admins_access module.key_vault["Dev"].module.admins``` , i.e effectively a simple rename of one level of a for_each.
